nebari_workflow_controller/utils.py

```diff
@@ -274,22 +312,24 @@ def mutate_template(
     spec_keep_portions,
     template,
 ):
-    for value, key in container_keep_portions:
-        if "container" not in template:
-            continue
+    target = "container" if "container" in template else "script" if "script" in template else None
```

Comment:
These are argo workflows template types. So I'd think we should have a variable called SUPPORTED_TEMPLATE_TYPES and then we check if each is in the template.
```
SUPPORTED_TEMPLATE_TYPES = {'container', 'script'}
```